faunadb-scala/src/load/scala/faunadb/FaunaClientLoadSpec.scala

```diff
@@ -187,6 +187,25 @@ class FaunaClientLoadSpec extends FixtureAsyncWordSpec with Matchers with ScalaF
         succeed
       }
     }
+
+    "fail to handle more than 100 concurrent streams on the same client" in { client =>
+      val maxConcurrentStreamCount = 100
+      // create collection
+      val collectionName = RandomGenerator.aRandomString
+      val setup = for {
+        _ <- client.query(CreateCollection(Obj("name" -> collectionName)))
+        createdDoc <- client.query(Create(Collection(collectionName), Obj("credentials" -> Obj("password" -> "abcdefg"))))
+        docRef = createdDoc("ref")
+        // create a first publisher to setup the connection that will be reused by all the other publishers (makes 101 streams)
+        _ <- client.stream(docRef)
+        publisherValues <- Future.traverse(List.fill(maxConcurrentStreamCount)(docRef))(ref => client.stream(ref))
```

Comment:
Following the [AAA](https://defragdev.com/blog/2014/08/07/the-fundamentals-of-unit-testing-arrange-act-assert.html) pattern with ScalaTest's [Asynchronous testing](https://www.scalatest.org/user_guide/async_testing) can be a bit tricky, but I was doing something like below:

```scala
def setup(): Future[Result[Value]] = {
  val collectionName = RandomGenerator.aRandomString
  for {
    _ <- client.query(CreateCollection(Obj("name" -> collectionName)))
    createdDoc <- client.query(Create(Collection(collectionName), Obj("credentials" -> Obj("password" -> "abcdefg"))))
  } yield createdDoc("ref").get
}

def run(docRef: Value): Future[List[Flow.Publisher[Value]]] = {
  for {
    // create a first publisher to setup the connection that will be reused by all the other publishers (makes 101 streams)
    _ <- client.stream(docRef)
    publisherValues <- Future.traverse(List.fill(maxConcurrentStreamCount)(docRef))(ref => client.stream(ref))
  } yield publisherValues
}

val result =
  for {
    docRef <- setup()
    result <- run(docRef)
  } yield result
```

Reply:
sorry I still have not taken the habit of using the AAA pattern.

Reply:
done!

Reply:
Yeah, that's fine, no worries! :)

The tests in the project in general are not following too many best practices, but we should try to revert that situation. I'd say we try to implement the pattern in this test case since the load test suite is rather new and small.

Alternatively, we could do something like below as well:

```scala
val result: Future[List[Flow.Publisher[Value]]] =
  for {
    // SETUP
    _ <- client.query(CreateCollection(Obj("name" -> collectionName)))
    createdDoc <- client.query(Create(Collection(collectionName), Obj("credentials" -> Obj("password" -> "abcdefg"))))
    docRef = createdDoc("ref")

    // RUN
    // create a first publisher to setup the connection that will be reused by all the other publishers
    _ <- client.stream(docRef)
    publisherValues <- Future.traverse(List.fill(maxConcurrentStreamCount - 1)(docRef))(ref => client.stream(ref))
    // adding 101st subscriber creates the error
    _ <- client.stream(docRef)

  } yield publisherValues

// VERIFY
recoverToExceptionIf[BadRequestException](result).map { e =>
  e.getMessage should include("the maximum number of streams has been reached for this client")
}
```

But, maybe there are too many comments together then? 🤔